app/javascript/onboarding/components/ProfileForm.jsx

```diff
@@ -50,23 +50,31 @@ export class ProfileForm extends Component {
   onSubmit() {
     const csrfToken = getContentOfToken('csrf-token');
     const { formValues, last_onboarding_page } = this.state;
+    const { username, ...newFormValues } = formValues;
     fetch('/onboarding_update', {
       method: 'PATCH',
       headers: {
         'X-CSRF-Token': csrfToken,
         'Content-Type': 'application/json',
       },
       body: JSON.stringify({
-        user: { last_onboarding_page },
-        profile: { ...formValues },
+        user: { last_onboarding_page, username },
+        profile: { ...newFormValues },
       }),
       credentials: 'same-origin',
```

Comment:
refactor (blocking): Use the `request` utility function instead.

Since you're modifying the fetch call, might as well migrate it to our `request` function. It can be imported like so `import { request } from '@utilities/http';`.

It handles stringifying the payload and all the headers you need by default.

```suggestion
    request('/onboarding_update', {
      method: 'PATCH',
      body: {
        user: { last_onboarding_page, username },
        profile: { ...newFormValues },
      },
```